pkg/cache/tas_flavor.go

```diff
@@ -108,12 +108,21 @@ func (c *TASFlavorCache) snapshotForNodes(log logr.Logger, nodes []corev1.Node,
 	snapshot := newTASFlavorSnapshot(log, c.Levels)
 	nodeToDomain := make(map[string]utiltas.TopologyDomainID)
 	for _, node := range nodes {
-		levelValues := utiltas.LevelValues(c.Levels, node.Labels)
-		capacity := resources.NewRequests(node.Status.Allocatable)
-		domainID := utiltas.DomainID(levelValues)
-		snapshot.levelValuesPerDomain[domainID] = levelValues
-		snapshot.addCapacity(domainID, capacity)
-		nodeToDomain[node.Name] = domainID
+		ready := false
+		for _, cond := range node.Status.Conditions {
+			// Only healthy and ready to accept pods nodes are considered for scheduling calculation
+			ready = (cond.Type == corev1.NodeReady && cond.Status == corev1.ConditionTrue)
```

Comment:
Also I think it would be great to move it to separate function. Like:

```
func IsNodeStatusConditionTrue(conditions []corev1.NodeCondition, conditionType corev1.NodeConditionType, conditionStatus corev1.ConditionStatus) bool {
	for _, cond := range conditions {
		if cond.Type == conditionType {
			return cond.Status == conditionStatus
		}
	}
	return false
}
```

Reply:
+1, then 
```
if !IsNodeStatusConditionTrue() {
  // Only healthy and ready to accept pods nodes are considered for scheduling calculation
  continue
}
```